util/design/lib/OtpMemMap.py

```diff
@@ -165,19 +165,34 @@ def _validate_mmap(config):
 
     offset = 0
     num_part = 0
+    part_index = OrderedDict()
     for part in config["partitions"]:
-        num_part += 1
         _validate_part(part, offset, key_names)
 
+        if part['name'] in part_index:
+            log.error('Partition name {} is not unique'.format(part['name']))
+            exit(1)
+
         # Loop over items within a partition
+        num_items = 0
+        item_index = OrderedDict()
         for item in part["items"]:
             _validate_item(item, offset)
+            if item['name'] in item_index:
+                log.error('Item name {} is not unique'.format(item['name']))
+                exit(1)
             log.info("> Item {} at offset {} with size {}".format(
                 item["name"], offset, item["size"]))
             offset += check_int(item["size"])
+            item_index.setdefault(item['name'], num_items)
```

Comment:
I'd consider just using `item_index[item['name']] = num_items` here. Line 181 already checks that you don't have a duplicate, so the code is probably easier to read that way.

Reply:
fixed!